src/sentry/api/endpoints/artifact_lookup.py

```diff
@@ -267,7 +216,21 @@ def get_artifact_bundles_containing_debug_id(
     )
 
 
-def get_release_artifacts(
+def get_artifact_bundles_containing_url(
+    project: Project, release_name: str, dist_name: Optional[str], url: str
+) -> Set[Tuple[int, datetime]]:
```

Comment:
Why does this return a set? Is the intention not that for each url we index the most recent bundle containing it?

Reply:
You are absolutely right. This is only for consistency with other similar functions, and downstream users. This will always return a single-element set, or an empty one.

Reply:
If we have duplicated entries, shouldn't we get the last one, we will break the consistency requirements if we don't do so?